manticore/native/memory.py

```diff
@@ -843,15 +838,15 @@ def mappings(self):
 
         return sorted(result)
 
-    def __str__(self):
+    def __str__(self) -> str:
```

Comment:
Does `mypy` not automatically expect `__str__` to return a string?

Reply:
No, `mypy` complains if you claim the return type of a `__str__` method is something other than `str`!

But, with the way we have `mypy` configured, only methods that have any type hints on them at all will be inspected by `mypy`.  So by adding the explicit return type annotation here, `mypy` will type-check the body of the method (which it otherwise wouldn't do).

Reply:
> the way we have mypy configured, only methods that have any type hints on them at all will be inspected by mypy

Barring a major refactor I think this works well for us, but I'm somewhat surprised that mypy doesn't override that for dunder methods, which really really should conform to the expected behavior even if it's not required of them.